subsys/bluetooth/audio/vocs_client.c

```diff
@@ -290,18 +295,28 @@ static uint8_t vcs_client_read_output_desc_cb(struct bt_conn *conn, uint8_t err,
 	BT_DBG("Inst %p: err: 0x%02X", inst, err);
 	inst->cli.busy = false;
 
+	if (cb_err) {
+		BT_DBG("Description read failed: %d", err);
+		if (inst->cli.cb && inst->cli.cb->description) {
+			inst->cli.cb->description(conn, inst, cb_err, NULL);
+		}
+		return BT_GATT_ITER_STOP;
+	}
+
```

Comment:
I think it is possible to remove the duplicated call to the callback, if one wants.

```
if (cb_err) {
        BT_DBG
} else if (data) {
        /* Handle the data */
}
[...]
if (inst.cli [...] ) {
        inst->cli.cb->description(conn, inst, (cb_err ? NULL, desc))
}

```

Reply:
It would be, but then I'd have to move lines 322-335 into an else clause to avoid setting the NULl terminator and attempting to print an invalid output description (or use a `goto` which I'm generally not a fan of).

Reply:
Ended up doing it anyhow :)